hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -339,26 +241,20 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         tbl.getParameters().remove(StatsSetupConst.COLUMN_STATS_ACCURATE);
       }
 
-      try {
-        if (hiveLockHeartbeat.future.isCancelled()
-            || hiveLockHeartbeat.encounteredException != null) {
-          throw new CommitFailedException(
-              "Failed to heartbeat for hive lock. %s",
-              hiveLockHeartbeat.encounteredException.getMessage());
-        }
+      lock.ensureActive();
 
+      try {
         persistTable(tbl, updateHiveTable);
-        if (hiveLockHeartbeat.future.isCancelled()
-            || hiveLockHeartbeat.encounteredException != null) {
-          throw new CommitStateUnknownException(
-              "Failed to heartbeat for hive lock while "
-                  + "committing changes. This can lead to a concurrent commit attempt be able to overwrite this commit. "
-                  + "Please check the commit history. If you are running into this issue, try reducing "
-                  + "iceberg.hive.lock-heartbeat-interval-ms.",
-              hiveLockHeartbeat.encounteredException);
-        }
+        lock.ensureActive();
 
         commitStatus = CommitStatus.SUCCESS;
+      } catch (LockException le) {
+        throw new CommitStateUnknownException(
```

Comment:
Hi @pvary, I think here we should set the `commitStatus` to `UNKNOWN` before throwing the `CommitStateUnknownException`. Otherwise the `commitStatus` is still `FAILURE` which could lead to the metadata being deleted in `cleanupMetadataAndUnlock`.

Reply:
Yea, I think thats a good catch

Reply:
Sounds right to me.
@ConeyLiu: would you mind creating a PR for fixing this?

Reply:
Thanks @szehon-ho @pvary, submit #7931